networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -220,42 +219,13 @@ def single_source_dijkstra_path_length(G, source, cutoff=None,
     single_source_dijkstra()
 
     """
-    push = heappush
-    pop = heappop
-    dist = {}  # dictionary of final distances
-    seen = {source: 0}
-    c = count()
-    fringe = []  # use heapq with (distance,label) tuples
-    push(fringe, (0, next(c), source))
-    while fringe:
-        (d, _, v) = pop(fringe)
-        if v in dist:
-            continue  # already searched this node.
-        dist[v] = d
-        # for ignore,w,edgedata in G.edges_iter(v,data=True):
-        # is about 30% slower than the following
-        if G.is_multigraph():
-            edata = []
-            for w, keydata in G[v].items():
-                minweight = min((dd.get(weight, 1)
-                                 for k, dd in keydata.items()))
-                edata.append((w, {weight: minweight}))
-        else:
-            edata = iter(G[v].items())
-
-        for w, edgedata in edata:
-            vw_dist = dist[v] + edgedata.get(weight, 1)
-            if cutoff is not None:
-                if vw_dist > cutoff:
-                    continue
-            if w in dist:
-                if vw_dist < dist[w]:
-                    raise ValueError('Contradictory paths found:',
-                                     'negative weights?')
-            elif w not in seen or vw_dist < seen[w]:
-                seen[w] = vw_dist
-                push(fringe, (vw_dist, next(c), w))
-    return dist
+    if G.is_multigraph():
+        get_weight = lambda u, v, data: min(eattr.get(weight, 1)
+            for eattr in data.values())
```

Comment:
Wrap the line immediately after `min(` instead of here. Please also review other instances of `get_weight`.